orchestra/interface_api/project_management/views.py

```diff
@@ -161,3 +165,29 @@ def set_project_status_api(request):
     except ProjectStatusError as e:
         raise BadRequest(e)
     return {'status': status, 'success': True}
+
+
+@project_management_api_view
+def staffbot_task(request):
+    data = load_encoded_json(request.body)
+    errors = {}
+    try:
+        task = Task.objects.get(id=data.get('task_id'))
+        request_cause = StaffBotRequest.RequestCause.USER.value
+        bot = StaffBot()
+        assignment = current_assignment(task)
+        is_restaff = assignment is not None
+        if is_restaff:
+            username = assignment.worker.user.username
+            bot.restaff(task.id, username,
+                        request_cause=request_cause)
+        else:
+            bot.staff(task.id, request_cause=request_cause)
+    except Exception as e:
+        errors['error'] = str(e)
```

Comment:
It looks like other endpoints in this file `raise BadRequest(e)` and don't try to capture the string for the exception in the return value. That way, we make it easier to debug since we're not gobbling up/hiding the stack trace.